go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -836,6 +841,36 @@ func (bsnc *BaseSecondaryNetworkController) buildUDNEgressSNAT(localPodSubnets [
 	return snats, nil
 }
 
+func (bsnc *BaseSecondaryNetworkController) ensureDHCP(pod *corev1.Pod, podAnnotation *util.PodAnnotation, lsp *nbdb.LogicalSwitchPort) error {
+
+	// Configure DHCP only for kubevirt VMs layer2 primary udn with subnets
+	if !kubevirt.VirtualMachineOwnsPod(pod) ||
+		!util.IsNetworkSegmentationSupportEnabled() ||
+		!bsnc.IsPrimaryNetwork() ||
+		bsnc.TopologyType() != types.Layer2Topology ||
+		len(bsnc.Subnets()) == 0 {
+		return nil
+	}
```

Comment:
I think this belongs in the caller. For a regular pod, I don't have to ensure DHCP flows at all. 

If you want this in the helper, I think we should rename the method. 

Also, please consider adding a helper for this fn. IIUC all it needs is the `netInfo`, along with the pod.